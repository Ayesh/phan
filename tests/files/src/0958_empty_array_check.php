<?php
/** @phan-file-suppress PhanPluginUseReturnValueNoopVoid, PhanUnreferencedFunction, PhanPluginDescriptionlessCommentOnFunction */
/** @phan-file-suppress PhanPluginCanUsePHP71Void, PhanPluginNoCommentOnFunction */

function getPossiblyEmptyArray(): array {
    return $_GET['foo'];
}

function test1(): void {
    $array = getPossiblyEmptyArray();
    if( count($array) === 0 ){
        '@phan-debug-var $array';
        return;
    }
    echo "done\n";
    '@phan-debug-var $array';
}

function test1b(): void {
    $array = getPossiblyEmptyArray();
    if( count($array) == 0 ){
        '@phan-debug-var $array';
        return;
    }
    echo "done\n";
    '@phan-debug-var $array';
}

function test2(): void {
    $array = getPossiblyEmptyArray();
    if( count($array) < 1 ){
        '@phan-debug-var $array';
        return;
    }
    echo "done\n";
    '@phan-debug-var $array';
}

function test3(): void {
    $array = getPossiblyEmptyArray();
    if( !count($array) ){
        '@phan-debug-var $array';
        return;
    }
    echo "done\n";
    '@phan-debug-var $array';
}

function test4(): void {
    $array = getPossiblyEmptyArray();
    if( !$array ){
        '@phan-debug-var $array';
        return;
    }
    echo "done\n";
    '@phan-debug-var $array';
}

function test5(): void {
    $array = getPossiblyEmptyArray();
    if( $array === [] ){
        '@phan-debug-var $array';
        return;
    }
    echo "done\n";
    '@phan-debug-var $array';
}
function test6(): void {
    $array = getPossiblyEmptyArray();
    if( $array == [] ){
        '@phan-debug-var $array';
        return;
    }
    echo "done\n";
    '@phan-debug-var $array';
}
