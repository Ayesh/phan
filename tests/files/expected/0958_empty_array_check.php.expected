%s:11 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type array{}(real=array{})
%s:15 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type non-empty-array<mixed,mixed>(real=non-empty-array<mixed,mixed>)
%s:21 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type array{}(real=array{})
%s:25 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type non-empty-array<mixed,mixed>(real=non-empty-array<mixed,mixed>)
%s:31 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type array{}(real=array{})
%s:35 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type non-empty-array<mixed,mixed>(real=non-empty-array<mixed,mixed>)
%s:41 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type array{}(real=array{})
%s:45 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type non-empty-array<mixed,mixed>(real=non-empty-array<mixed,mixed>)
%s:51 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type array{}(real=array{})
%s:55 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type non-empty-array<mixed,mixed>(real=non-empty-array<mixed,mixed>)
%s:61 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type array{}(real=array{})
%s:65 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type non-empty-array<mixed,mixed>(real=non-empty-array<mixed,mixed>)
%s:70 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type array{}(real=array{})
%s:74 PhanDebugAnnotation @phan-debug-var requested for variable $array - it has union type non-empty-array<mixed,mixed>(real=non-empty-array<mixed,mixed>)
